crates/video/src/stream.rs

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+// Null Backend does not use all stream capabilities
+#![cfg_attr(not(any(feature)), allow(dead_code))]
+
+use enumn::N;
+use std::{
+    collections::HashMap,
+    fs::File,
+    future::Future,
+    os::fd::AsRawFd,
+    path::Path,
+    pin::Pin,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc, RwLock,
+    },
+    task::{Context, Poll, Waker},
+};
+
+use crate::{
+    vhu_video::{Result, VuVideoError},
+    video::{BufferFlags, Format, MemoryType, QueueType},
+};
+
+#[repr(C)]
+#[derive(Clone, Debug, Default)]
+pub struct ResourcePlane {
+    pub offset: u32,
+    pub address: u64,
+    pub length: u32,
+}
+
+#[repr(C)]
+#[derive(Clone, Debug, N, Eq, PartialEq)]
+pub enum ResourceState {
+    Created = 1,
+    Queried,
+    Queued,
+    Ready,
+}
+
+impl Default for ResourceState {
+    fn default() -> Self {
+        Self::Created
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+pub struct SharedResourceState {
+    value: ResourceState,
+    waker: Option<Waker>,
+}
+
+impl SharedResourceState {
+    pub fn is_ready(&self) -> bool {
+        self.value == ResourceState::Ready
```

Comment:
```suggestion
        matches!(self.value, ResourceState::Ready)
```
is the rust-y way i guess